crates/matrix-sdk/src/media.rs

```diff
@@ -198,49 +203,51 @@ impl Media {
 
         let inferred_extension = mime2ext::mime2ext(content_type);
 
-        let body_path = body.as_ref().map(Path::new);
-        let filename = body_path.and_then(|f| f.file_name().and_then(|f| f.to_str()));
-        let filename_with_extension = body_path.and_then(|f| {
+        let filename_as_path = filename.as_ref().map(Path::new);
+        let filename_without_extension =
+            filename_as_path.and_then(|f| f.file_name().and_then(|f| f.to_str()));
+        let filename_with_extension = filename_as_path.and_then(|f| {
             if f.extension().is_some() {
                 f.file_name().and_then(|f| f.to_str())
```

Comment:
ditto: same here

Actually, since you have two code paths that deal with `if the filename as path exists`, use `if let Some(filename_path)` instead, and return a pair tuple?

Reply:
Ah yeah, that sounds more sane, will update 👍

Reply:
Updated, thanks that's much clearer.